Refactor typeofEquals usage and remove isInstanceOfDomType

srujzs opened the pull request:
Closes https://github.com/dart-lang/web/issues/77

typeofEquals now takes a String and returns a bool, and instanceOfString is the canonical replacement for isInstanceOfDomType. Removes now unneeded test as well. Updates min SDK version.

srujzs commented:
This will have to wait until we publish a Dart dev version >= this minimum.

kevmoo commented on pubspec.yaml:
```diff
@@ -6,7 +6,7 @@ version: 0.2.1-beta
 
 environment:
-  sdk: ">=3.2.0-157.0.dev <4.0.0"
+  sdk: ">=3.2.0-194.0.dev <4.0.0"
```
double check pubspec version and changelog version

srujzs replied:
Done, thanks!

sigmundch reviewed (approved):
Overall looks good, but I'd also be OK keeping a test like the one deleted in helpers_test. That's because, while technically the `instnaceOfString` should be tested by the SDK, there is still value in having a test here that ensures that it works for package:web types.

srujzs commented:
> Overall looks good, but I'd also be OK keeping a test like the one deleted in helpers_test. That's because, while technically the `instnaceOfString` should be tested by the SDK, there is still value in having a test here that ensures that it works for package:web types.

Fair point, added it back and refactored to account for `instanceOfString`.

sigmundch reviewed (approved):
Thanks! :+1:
